src/gas/mod.rs

```diff
@@ -501,6 +502,20 @@ pub fn inject_gas_counter<R: Rules>(
 			elements::Section::Start(start_idx) => {
 				if *start_idx >= gas_func { *start_idx += 1}
 			},
+			elements::Section::Name(section) => {
+				if let Some(funcs) = section.functions_mut() {
+					let mut new_names = elements::IndexMap::<String>::default();
```

Comment:
Please implement it without re-allocating the whole thing. Like for the other function sections.

Reply:
The `xx_mut()` in other sections are `&mut Vec<T>`, and the `functions_mut()` can not be modified directly since it returns an `IndexMap`.

Reply:
Okay by looking closer at the data structure I realized why it is not easily mutable. Still, we don't want to add such a heavy operation to the on-chain gas instrumentation.

Additionally, `parse_names()` is never called during gas injection. So this code is never hit. How did you test it?

Reply:
Well, we `parse_names()` after getting wasm code from storage, maybe we can add the param `ProcessNames` to `prepare_contract()`?